DigitalLearningSolutions.Web/Scripts/trackingSystem/addNewCentreCourseSelectCourse.ts

```diff
@@ -0,0 +1,159 @@
+}
+
+function filterElements(
+  searchableElements: ISearchableElement[],
+  appliedFilter: IAppliedFilter,
+): ISearchableElement[] {
+  return searchableElements.filter(
+    (element) => doesElementMatchFilterValue(element, appliedFilter.filterValue),
+  );
+}
+
+function displaySearchableElements(searchableElements: ISearchableElement[]): void {
+  const searchableElementsContainer = document.getElementById('searchable-elements');
+  if (!searchableElementsContainer) {
+    return;
+  }
+
+  const defaultOption = document.getElementById('default-option');
+  if (!defaultOption) {
+    return;
+  }
+  defaultOption.innerText = searchableElements.length === 0 ? 'No matching courses' : 'Select a course';
+
+  searchableElementsContainer.textContent = '';
+  searchableElementsContainer.appendChild(defaultOption);
+  searchableElements.forEach(
+    (searchableElement) => searchableElementsContainer.appendChild(searchableElement.element),
+  );
+
+  // This is required to polyfill the new elements in IE
+  Details();
+}
+
+function getCategoryAndTopicFilterByAndUpdateHiddenInputs() {
+  const categoryFilterElement = <HTMLSelectElement>document.getElementById('CategoryName');
+  const categoryFilterValue = categoryFilterElement.value;
+
+  const topicFilterElement = <HTMLSelectElement>document.getElementById('CourseTopic');
+  const topicFilterValue = topicFilterElement.value;
+
+  if (isNullOrEmpty(categoryFilterValue) && isNullOrEmpty(topicFilterValue)) {
+    return null;
+  }
+
+  if (isNullOrEmpty(categoryFilterValue)) {
+    updateFilterByHiddenInput(topicHiddenInputName, topicFilterValue);
+    return topicFilterValue;
+  }
+
+  if (isNullOrEmpty(topicFilterValue)) {
+    updateFilterByHiddenInput(categoryHiddenInputName, categoryFilterValue);
+    return categoryFilterValue;
+  }
+
+  updateFilterByHiddenInput(categoryHiddenInputName, categoryFilterValue);
+  updateFilterByHiddenInput(topicHiddenInputName, topicFilterValue);
+
+  return topicFilterValue + filterSeparator + categoryFilterValue;
+}
+
+function updateFilterByHiddenInput(elementName: string, newFilter: string): void {
+  const filterByElements = Array.from(document.getElementsByName(elementName));
+
+  filterByElements.forEach((filterByElement) => {
+    const hiddenInput = <HTMLInputElement>filterByElement;
+    hiddenInput.value = newFilter;
+    sendBrowserAgnosticEvent(hiddenInput, 'change');
+  });
+}
+
+function doesElementMatchFilterValue(
+  searchableElement: ISearchableElement,
+  filterValue: string,
+): boolean {
+  const categoryFilterValue = searchableElement.element.getAttribute('category-filter-value')?.trim() ?? '';
+  const topicFilterValue = searchableElement.element.getAttribute('topic-filter-value')?.trim() ?? '';
+  return categoryFilterValue === filterValue || topicFilterValue === filterValue;
+}
+
+function isNullOrEmpty(
+  filterValue: string,
+): boolean {
+  return typeof filterValue === undefined || !filterValue;
+}

```

Comment:
Might be worth moving this into common.ts since it might be useful elsewhere

Reply:
Moved :)